hack/generator/pkg/config/export_filter.go

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package config
+
+// exportFilterAction defines the possible actions that should happen for types matching the filter
+type exportFilterAction string
+
+const (
+	// exportFilterInclude indicates that any type matched by the filter should be exported to disk by the generator
+	exportFilterInclude exportFilterAction = "include"
+
+	// exportFilterExclude indicates that any type matched by the filter should not be exported as a struct.
+	// This skips generation of types but may not prevent references to the type by other structs (which
+	// will cause compiler errors).
+	exportFilterExclude exportFilterAction = "exclude"
+)
+
+// A ExportFilter is used to control which types should be exported by the generator
+type ExportFilter struct {
+	Action      exportFilterAction
```

Comment:
It seems a bit weird for the type and values for this field be unexported - how will client code construct `ExportFilter`s?

Reply:
Doh, it's only used in this package so I thought I could make it private, but realizing now that doesn't work because yaml would deserialize to it properly.

Reply:
Yeah, I was thinking that this is only used internally by `Configuration` (which is true), but it doesn't really make sense to make this internal without making a whole lot of other stuff internal too...

reverting it to be public like I had it before.